background/storage.ts

```diff
@@ -0,0 +1,115 @@
+import browser from "webextension-polyfill"
+
+import { config } from "~config"
+import { ObservableStorage } from "~packages/storage/observable"
+import type { HexString } from "~typing"
+
+let storage: ObservableStorage<State>
+
+export async function getStorage() {
+  if (!storage) {
+    // TODO: Check browser.runtime.lastError
+    const state = await browser.storage.local.get(null)
+    storage = new ObservableStorage<State>(state as State)
+    storage.subscribe(async (state) => {
+      await browser.storage.local.set(state)
+    })
+    // TODO: Only for development at this moment. Remove following when getting to production.
+    // Enable only network specified in env
+    storage.set(
+      {
+        network: {
+          [config.chainId]: {
+            active: true,
+            chainId: config.chainId,
+            nodeRpcUrl: config.nodeRpcUrl,
+            bundlerRpcUrl: config.bundlerRpcUrl,
+            account: {
+              ...(config.simpleAccountAddress && {
+                [config.simpleAccountAddress]: {
+                  type: AccountType.SimpleAccount,
+                  address: config.simpleAccountAddress,
+                  ownerPrivateKey: config.simpleAccountOwnerPrivateKey
+                }
+              }),
+              ...(config.passkeyAccountAddress && {
+                [config.passkeyAccountAddress]: {
+                  type: AccountType.PasskeyAccount,
+                  address: config.passkeyAccountAddress
+                }
+              })
+            },
+            paymaster: {
+              ...(config.verifyingPaymasterAddress && {
+                [config.verifyingPaymasterAddress]: {
+                  type: PaymasterType.VerifyingPaymaster,
+                  address: config.verifyingPaymasterAddress,
+                  ownerPrivateKey: config.verifyingPaymasterOwnerPrivateKey
+                }
+              })
+            }
+          }
+        }
+      },
+      { override: true }
+    )
+  }
```

Comment:
I think using named parameter types for readability once the `State` type is more certain, despite potential complexity.